### DISCUSSION TITLE
[bug]: lnd fails to generate tls certificate

### DISCUSSION THREAD

**Discussion by ekimber** [lightningnetwork/lnd#9193] [Troubleshooting] [answered]
### Background

lnd fails to start because tls certificate can't be generated

> error setting cert before unlock: unable to generate or renew TLS certificate: route ip+net: netlinkrib: address family not supported by protocol

### Your environment

* `lnd` 0.18.3-beta
* Linux 5.4.281 1-NixOS SMP Sat Jul 27 08:38:33 UTC 2024 x86_64 GNU/Linux
* bitcoind 27.1

### Steps to reproduce

Try to start the lnd process

**Comment by saubyk** [comment]
This error "route ip+net: netlinkrib: address family not supported by protocol" seems to be IPV6 related

Is IPV6 disabled on your device?

**Reply by ekimber** [reply]
No, IPv6 is enabled

**Reply by saubyk** [reply]
Can you provide some detail of your hardware config and networking setup? Is it just a regular setup or you have any specific networking nuance here like specific firewall setting etc?
Is this the first time you're running LND?

**Reply by ekimber** [reply, marked as answer]
Hi there, thanks for your time. I have solved the problem, it was that the netlink address family was disallowed by a systemd security rule. I have removed the rule and it works. Sorry for any inconvenience
